_blog/2018/soft-multitenancy.md

```diff
@@ -0,0 +1,311 @@
+metadata:
+  name: access-all-istio-system1
+  namespace: istio-system1
+subjects:
+- kind: User
+  name: sales-admin
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: Role
+  name: ns-access-for-sales-admin-istio-system1
+  apiGroup: rbac.authorization.k8s.io
+
+```
+
+### Watching specific namespaces for service discovery
+In addition to creating RBAC rules limiting the tenant administrator access to a specific Istio
+control plane, the istio manifest must be updated to specify the application namespace that
+Pilot should watch for creation of its xDS cache. This is done by starting the Pilot component
+with the additional command line arguments,  `--appNamespace, ns-1`.  Where *ns-1* is the
+namespace that the tenant’s application will be deployed in. An example snippet from the
+istio-system1.yaml file is included below.
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: istio-pilot
+  namespace: istio-system1
+  annotations:
+    sidecar.istio.io/inject: "false"
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        istio: pilot
+    spec:
+      serviceAccountName: istio-pilot-service-account
+      containers:
+      - name: discovery
+        image: docker.io/<user ID>/pilot:<tag>
+        imagePullPolicy: IfNotPresent
+        args: ["discovery", "-v", "2", "--admission-service", "istio-pilot", "--appNamespace", "ns-1"]
+        ports:
+        - containerPort: 8080
+        - containerPort: 443
+
+```
+
+### Deploying the tenant application in a namespace
+Now that the cluster administrator has created the tenant's namespace (ex. *istio-system1*) and
+the Pilot's service discovery has been configured to watch for a specific application
+namespace (ex. *ns-1*), create the application manifests to deploy in that tenant's specific
+namespace. Example:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ns-1
+```
+And add the namespace reference to each resource type included in the applications manifest
+file.  For example:
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: details
+  labels:
+    app: details
+  namespace: ns-1
+```
+Although not shown, the application namespaces will also have RBAC settings limiting access
+to certain resources. These RBAC settings could be set by the cluster administator and/or
+the tenant administator.
+
+### Using Istioctl commands in a multitenant environment
+When defining [route rules]({{home}}/docs/reference/config/istio.routing.v1alpha1.html#RouteRule)
+or [destination policies]({{home}}/docs/reference/config/istio.routing.v1alpha1.html#DestinationPolicy),
+it is necessary to ensure that the `istioctl` command is scoped to
+the namespace the Istio control plane is running in to ensure the resource is created
+in the proper namespace. Additionally, the rule itself must be scoped to the tenant's namespace
+so that it will be applied properly to that tenant's mesh.  The "-i" option is used to create
```

Comment:
Here you use "-i" but elsewhere you've used *-i* to denote options. Probably want to make everything consistent.

Reply:
done (i think :-) )